x-pack/plugins/cases/public/components/custom_fields/types.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type React from 'react';
+
+export enum CustomFieldTypes {
+  TEXT = 'text',
+  TOGGLE = 'toggle',
+}
+
+export interface ConfigureCustomFields {
+  key: string;
+  label: string;
+  type: CustomFieldTypes;
+  required: boolean;
+}
+export interface CustomFieldBuildType {
+  ConfigurePage: React.FC;
+}
+
+export type CustomFieldBuilder = () => {
+  build: () => CustomFieldBuildType[];
```

Comment:
I don't think we need to return an array, right? There will only be one configuration object for each field type.